pkg/materialize/sink_kafka.go

```diff
@@ -130,8 +130,45 @@ func (b *SinkKafkaBuilder) Create() error {
 		if b.format.Avro.SchemaRegistryConnection.Name != "" {
 			q.WriteString(fmt.Sprintf(` FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION %s`, b.format.Avro.SchemaRegistryConnection.QualifiedName()))
 		}
+
+		// CSR Connection Options
+		var v = []string{}
 		if b.format.Avro.AvroValueFullname != "" && b.format.Avro.AvroKeyFullname != "" {
-			q.WriteString(fmt.Sprintf(` WITH (AVRO KEY FULLNAME %s AVRO VALUE FULLNAME %s)`, QuoteString(b.format.Avro.AvroKeyFullname), QuoteString(b.format.Avro.AvroValueFullname)))
+			v = append(v, fmt.Sprintf(`AVRO KEY FULLNAME %s AVRO VALUE FULLNAME %s`,
+				QuoteString(b.format.Avro.AvroKeyFullname),
+				QuoteString(b.format.Avro.AvroValueFullname)),
+			)
+		}
+
+		// Doc Type
+		if b.format.Avro.DocType.Object.Name != "" {
+			c := strings.Builder{}
+			if b.format.Avro.DocType.Key {
+				c.WriteString("KEY ")
+			} else if b.format.Avro.DocType.Value {
+				c.WriteString("VALUE ")
+			}
+			c.WriteString(fmt.Sprintf("DOC ON TYPE %[1]s = %[2]s",
+				b.format.Avro.DocType.Object.QualifiedName(),
```

Comment:
Based on docs, unsure if we need to also support aliases for the object reference. Or if it should just always make the connection.